osu.Game.Rulesets.Catch/Objects/Drawables/DrawableDroplet.cs

```diff
@@ -28,15 +28,22 @@ private void load()
                 _ => new DropletPiece());
         }
 
-        protected override void UpdateInitialTransforms()
+        private float startRotation;
+
+        protected override void OnApply()
         {
-            base.UpdateInitialTransforms();
+            base.OnApply();
 
             // roughly matches osu-stable
-            float startRotation = RandomSingle(1) * 20;
-            double duration = HitObject.TimePreempt + 2000;
+            startRotation = RandomSingle(1) * 20;
+        }
+
+        protected override void Update()
+        {
+            base.Update();
 
-            ScalingContainer.RotateTo(startRotation).RotateTo(startRotation + 720, duration);
+            double preemptProgress = (Time.Current - (HitObject.StartTime - InitialLifetimeOffset)) / (HitObject.TimePreempt + 2000);
```

Comment:
Why is this clamped in `DrawableBanana` (albeit only from above to 1, rather than from both sides), and not here? `Interpolation.Lerp()` does no clamping on its own, so it will *extra*polate when given arguments outside [0;1].

Reply:
The clamping in `DrawableBanana` is because we assume that scale shouldn't continue to decrease after reaching the final value. For rotation, I want it to continue in both directions for sanity reasons (assume it's always spinning).

This is the same reason clamping is only applied to max not min for scale – to allow extrapolating values for the range which previously didn't animate.

I can add inline comments if this isn't clear.

Reply:
Inline comments could be useful because all that makes sense but is also very non-obvious.

Reply:
Will do. Also having talked through the above I realised that bananas will look wrong if you don't catch them, will fix that.